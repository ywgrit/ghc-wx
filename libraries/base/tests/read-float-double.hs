-- Test edge cases fixed in
-- https://gitlab.haskell.org/ghc/ghc/-/merge_requests/10933

main :: IO ()
main = do
  let
    lo = toInteger (minBound :: Int)
    hi = toInteger (maxBound :: Int)
    readDouble = read :: String -> Double
    readFloat = read :: String -> Float

  print $ readDouble ("1e" <> show lo)
  print $ readDouble ("1e" <> show (lo - 1))
  print $ readDouble ("0.01e" <> show lo)
  print $ readDouble ("10e" <> show hi)

  print $ readFloat ("1e" <> show lo)
  print $ readFloat ("1e" <> show (lo - 1))
  print $ readFloat ("0.01e" <> show lo)
  print $ readFloat ("10e" <> show hi)
