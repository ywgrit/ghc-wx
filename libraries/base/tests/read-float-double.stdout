0.0
0.0
0.0
Infinity
0.0
0.0
0.0
Infinity
